xwiki-platform-core/xwiki-platform-realtime/xwiki-platform-realtime-default/src/main/java/org/xwiki/realtime/XWikiRealtimeDocumentLockEditConfirmationChecker.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.realtime;
+
+import java.util.Optional;
+
+import javax.annotation.Priority;
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.inject.Provider;
+import javax.inject.Singleton;
+
+import org.xwiki.model.validation.edit.EditConfirmationCheckerResult;
+import org.xwiki.model.validation.edit.XWikiDocumentLockEditConfirmationChecker;
+
+import com.xpn.xwiki.XWikiContext;
+import com.xpn.xwiki.doc.XWikiDocument;
+
+/**
+ * Checks if a a realtime session is running, if not: check if the document is locked.
+ *
+ * @version $Id$
+ * @since 16.2.0
+ * @since 16.3.0RC1
+ */
+@Singleton
+@Named("documentLock")
```

Comment:
@manuelleduc no convention for the names of those components? Sounds like this one should be more `realtimeDocumentLock`

Reply:
This is actually an override of https://github.com/xwiki/xwiki-platform/blob/543fd0bfb897b570008851ed0b7c21594548c13f/xwiki-platform-core/xwiki-platform-oldcore/src/main/java/org/xwiki/model/validation/edit/XWikiDocumentLockEditConfirmationChecker.java#L46
@douakli might be a good idea to extract `"documentLock"` to a constant.
Same for the priority actually.